src/org/way2it/data_structures/list/List.java

```diff
@@ -14,30 +16,61 @@ public List() {
     // Should add new value to the end of the list, increment length and extend array by x2 if needed
     public void add(String value) {
         // TODO implement me
+        if (length == values.length) {
+            String[] temp = values;
+            this.values = new String[values.length+1];
+            System.arraycopy(temp,0,values, 0, temp.length);
+        }
+        values[length] = value;
+        length++;
     }
 
     // Should add new value at the specified index, moving other values to the right,
     // increment length and extend array by x2 if needed
     public void addAtIndex(String value, int index) {
         // TODO implement me
+        if (length == values.length) {
+            String[] temp = values;
+            this.values = new String[values.length+1];
```

Comment:
Same as previous comment. Also, it would be great to extract array-extending logic from this two methods to a new separate method. It will help to avoid code duplication